react-app-drawer-component/my-app/src/AppDrawer.js

```diff
@@ -0,0 +1,33 @@
+import { FaBars } from 'react-icons/fa';
+import { useState } from 'react';
+import './AppDrawer.css';
+
+export default function AppDrawer() {
+  const [showModal, setShowModal] = useState(false);
+
+  function handleSpanClick(e) {
+    setShowModal(!showModal);
+    if (e.target.className === 'modal') {
+      showModal(!false);
+    }
+  };
+
+
+  return (
+    <>
+      <span className='extra' onClick={handleSpanClick}><FaBars /></span>
+      {showModal && (
+      <div className="modal">
+        <div className="modal-content">
+          <h1 >Choose a Game</h1>
+          <h2 onClick={handleSpanClick}>The Legend of Zelda</h2>
```

Comment:
when you have a list of repeating elements like this with only small variations, you want to extract them out as a separate component and import.